rusk/tests/common/mod.rs

```diff
@@ -8,32 +8,38 @@ pub mod encoding;
 #[cfg(not(target_os = "windows"))]
 pub mod unix;
 
-use super::SOCKET_PATH;
+use super::{new_socket_path, SOCKET_DIR};
+
+use futures::future::BoxFuture;
+use std::convert::TryFrom;
+use std::path::{Path, PathBuf};
+use std::task::{Context, Poll};
+use std::{fs, io};
+
 use futures::TryFutureExt;
 use rusk::services::network::NetworkServer;
 use rusk::services::network::RuskNetwork;
 use rusk::services::pki::KeysServer;
+use rusk::services::prover::ProverServer;
 use rusk::Rusk;
-use std::convert::TryFrom;
 use test_context::AsyncTestContext;
-use tokio::net::UnixListener;
-use tokio::net::UnixStream;
+use tokio::net::{UnixListener, UnixStream};
 use tonic::transport::{Channel, Endpoint, Server, Uri};
-use tower::service_fn;
+use tower::Service;
 use tracing::{subscriber, Level};
 use tracing_subscriber::fmt::Subscriber;
 
 pub struct TestContext {
     pub channel: Channel,
+    socket_path: PathBuf,
 }
 
 #[async_trait::async_trait]
 impl AsyncTestContext for TestContext {
     async fn setup() -> TestContext {
-        // First of all let's remove old stucked socket path
         // This is required because the tear_down functions is not called if any
-        // test panics
-        let _ = std::fs::remove_file(&*SOCKET_PATH);
+        // test panics.
+        let _ = delete_all_possible(&*SOCKET_DIR).expect("existing directory");
```

Comment:
I noticed that `SOCKET_DIR` also take care of its creation. I think it's wrong. Either `SOCKET_DIR` delete itself properly because the `tempdir` crate, or if it doesn't it should be just a `&'static str` that is used in the tests to create / delete the folder.

If that is the case, then `delete_all_possible` should be just a `setup_socket_dir` (or similar) so that instead of removing manually each possible entry of the folder, it just remove the folder directly and all the contents, and then it creates is again from scratch, empty.

Something like this:
```rust
function setup_socket_dir<P: AsRef<Path>>(path: P) -> Result<(), io::Error> {
    fs::remove_dir_all("antani").or_else(|e| {
        if e.kind() == io::ErrorKind::NotFound {
            Ok(())
        } else {
            Err(e)
        }
    })?;

    // Not sure if two levels are necessary,
    // I would probably just have:
    //
    // temp_dir().join("rusk_socket_dir_test")
    //
    // Or similar.
    let tmp_dir = temp_dir().join(".rusk").join(".tmp_test");
    fs::create_dir_all(tmp_dir.clone())

    Ok(())
}
```

Reply:
Seems reasonable. I agree the setup of the directory should be done in `TestContext`, like everything else. The issue here is it shouldn't really error on a problem with "removing everything", since another test might be keeping a UDS busy. I'll change it.